bot/src/main/kotlin/me/melijn/bot/commands/ReminderCommand.kt

```diff
@@ -0,0 +1,170 @@
+import me.melijn.apkordex.command.KordExtension
+import me.melijn.bot.database.manager.ReminderManager
+import me.melijn.bot.services.ReminderService
+import me.melijn.bot.utils.KordExUtils.atLeast
+import me.melijn.bot.utils.KordExUtils.bail
+import me.melijn.bot.utils.KordExUtils.tr
+import me.melijn.bot.utils.intRanges
+import me.melijn.gen.RemindersData
+import me.melijn.kordkommons.utils.escapeCodeBlock
+import org.koin.core.component.inject
+import java.time.Instant
+import java.time.ZonedDateTime
+import java.time.temporal.ChronoUnit
+import java.util.concurrent.CancellationException
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.toJavaDuration
+
+@KordExtension
+class ReminderCommand : Extension() {
+    override val name: String = "reminder"
+
+    val reminderManager by inject<ReminderManager>()
+    val service by inject<ReminderService>()
+
+    override suspend fun setup() {
+        publicSlashCommand {
+            name = "reminder"
+            description = "Manages reminders"
+
+            ephemeralSubCommand(::ReminderAddArgs) {
+                name = "add"
+                description = "Creates a reminder"
+
+                action {
+                    val data = RemindersData(
+                        user.idLong,
+                        (Instant.now() + arguments.remindAt.toJavaDuration()).toKotlinInstant(),
+                        arguments.text
+                    )
+                    reminderManager.store(data)
+                    service.waitingJob.cancel(CancellationException("new reminder state, recheck first"))
+
+                    respond {
+                        val timeStamp = data.moment.toDiscord(TimestampType.LongDateTime)
+                        val text = data.reminderText.escapeCodeBlock()
+                        content = tr("reminders.added", timeStamp, text)
+                    }
+                }
+            }
+
+            ephemeralSubCommand(::ReminderRemoveArgs) {
+                name = "remove"
+                description = "Removes a reminder"
+
+                action {
+                    val reminders = reminderManager.getByUserIndex(this.user.idLong)
+                        .sortedBy { it.moment }
+                    if (reminders.isEmpty()) {
+                        respond {
+                            content = tr("reminders.list.empty")
+                        }
+                        return@action
+                    }
+
+                    val toRemove = mutableSetOf<RemindersData>()
+                    for (i in arguments.indices.list) {
+                        val iter = i.iterator()
+                        var index = iter.nextInt()
+                        toRemove.add(reminders[index])
+                        while (index < reminders.size && iter.hasNext()) {
+                            index = iter.nextInt()
+                            toRemove.add(reminders[index])
+                        }
+                    }
```

Comment:
```suggestion
                    val toRemove = arguments.indices.list.flatten().map { reminders[it] }.toMutableSet()
```
untested, but what the hell is this logic